### ISSUE TITLE
Error on computing the Jacobian of a vector-valued function when output and input vectors have different dimensions

### ISSUE THREAD

**Issue by b4sgren** [autodiff/autodiff#24]
I have a vector valued function that returns a 3 vector but I get an error when I take the gradient with respect to the input vector which has 12 variables. Can autodiff do a non-square jacobian of a vector valued function?

~~~c++
VectorXdual f(const VectorXdual& x)
{
    VectorXdual res = VectorXdual::Zero(3);
    dual theta = 0.0;

    for(int i(0); i < x.rows()/3; i++)
    {
        Eigen::Matrix3dual R;
        R << cos(theta), -sin(theta), 0, sin(theta), cos(theta), 0, 0, 0, 1;
        res.head(3) += R * x.segment(3*i, 3);
        res(2) = wrap(res(2));
        theta = res(2);
    }

    return res;
}

int main()
{
    //Vector function. Doesn't like it not being square
    VectorXdual u = f(x);
    Eigen::MatrixXd J = autodiff::jacobian(f, u, x);

    std::cout << "dux_dx = \n" << dux_dx << "\n\n";
    std::cout << "duy_dx = \n" << duy_dx << "\n\n";
    std::cout << "duphi_dx = \n" << duphi_dx << "\n\n";

    std::cout << "J = \n" << J << "\n\n";

    return 0;
}
~~~

**Comment by allanleal** [comment]
Hi! What is the error? Could you please post an example that compiles, so I can test here?

**Comment by allanleal** [comment]
OK, I can confirm there was a bug when output and input had different dimensions. Just pushed a bug fix. Once this is merged, you should update to version `0.4.1`. Thanks for reporting! ;)

**Comment by allanleal** [comment]
Fixed in version 0.4.1
